src/Sulu/Bundle/SecurityBundle/AccessControl/AccessControlQueryEnhancer.php

```diff
@@ -115,12 +114,12 @@ private function enhanceQueryWithAccessControl(
             'WITH',
             $accessClassCondition . ' AND ' . $this->getEntityIdCondition($entityClass, 'entity', $entityIdField)
         );
-        $subQueryBuilder->leftJoin('accessControl.role', 'role', 'WITH', 'role.system = :system');
+        $subQueryBuilder->leftJoin('accessControl.role', 'role');
         $subQueryBuilder->andWhere(
             'BIT_AND(accessControl.permissions, :permission) <> :permission AND accessControl.permissions IS NOT NULL'
         );
 
-        $subQueryBuilder->andWhere('role.id IN(:roleIds) OR role.id IS NULL');
+        $subQueryBuilder->andWhere('role.id IN(:roleIds) AND role.system = :system');
```

Comment:
As `:roleIds` is already filtered by `system` in `getUserRoleIds` the ` AND role.system = :system` is not required longer  here.

Reply:
changed 🙂